pallets/loans/src/lib.rs

```diff
@@ -433,6 +435,13 @@ pub mod pallet {
 		}
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			migrations::migrate_from_v2_to_v3::<T>()
+		}
+	}
+
```

Comment:
Can we have this as we usually do and make this a struct that is feed manually to the runtime?

Reply:
But no blocker.

Reply:
Just hard to see what is already getting migrated.

Reply:
I did so because it seemed to be the canonical way to migrate things that belong strictly to the internals of the pallet.

Actually, there is an annoying log message in the node that tells you that loans are not migrated when it is done outside the `hooks`. Anyway, I have no strong opinion regarding this

Reply:
I haven't seen a migration triggered this way yet. So I would prefer going the typical `OnRuntimeUpgrade` path such we can ensure storage can be decoded properly beforehand and afterwards.

Reply:
2 vs 1. Then I change it for sure 😆

For reference, I saw this here in the official docs here: https://docs.substrate.io/reference/how-to-guides/storage-migrations/basic-storage-migration/

Reply:
Thanks for the reference! Honestly, it's a simple migration. Just keep as is and we can move on.

Reply:
👍🏻

Reply:
FYI, the above way generates a first-line log in the following way by `frame-support`:

```
2024-02-26 09:15:02.252  INFO main runtime::frame-support: ⚠️ Loans declares internal migrations (which *might* execute). On-chain `StorageVersion(2)` vs current storage version `StorageVersion(3)`
2024-02-26 09:15:02.252  INFO main pallet_loans::migrations: Loans: Starting migration v3
2024-02-26 09:15:02.252  INFO main pallet_loans::migrations: Loans: Migrated 1 pools
```